src/librustc_typeck/check/_match.rs

```diff
@@ -659,6 +659,12 @@ pub fn check_pat_enum<'a, 'tcx>(pcx: &pat_ctxt<'a, 'tcx>,
     let report_bad_struct_kind = |is_warning| {
         bad_struct_kind_err(tcx.sess, pat.span, path, is_warning);
         if is_warning {
+            // Boo! Too painful to attach this to the actual warning,
```

Comment:
(I think I should have pushed back harder here.   But then again I didn't expect bors to take four days to get to this)